cmd/svcat/parameters/parameters.go

```diff
@@ -56,19 +57,45 @@ func ParseVariableAssignments(params []string) (map[string]interface{}, error) {
 		}
 		value := strings.TrimSpace(parts[1])
 
+		if strings.ContainsAny(variable, ".") {
+			//check if params form is -p xxx.xx
+			r, _ := regexp.Compile(`[^\s\.]+\.[^\s\.]+(\.[^\s\.]+)*`)
+			if r.MatchString(variable) == true {
+				dotKeys = strings.Split(variable, ".")
+				variable = dotKeys[0]
+			} else {
+				return nil, fmt.Errorf("invalid parameter (%s), must be in x.x format", variable)
+			}
+		}
+
 		storedValue, ok := variables[variable]
 		// Logic to add new value to map variables:
-		// if variable DNE: add pair to variables as variable:value
-		// if variable exists in form of variable:value, create array to hold old value&new value
-		// if variable exists in form variable:[some values], append new value to existing array
+		// if variable DNE and not exploding params -> variable:value
+		// if variable DNE and exploding params -> variable: map[somekey:somevalue]
+		// Logic to add value to variables where variable already exists in variables:
+		// if variable:value -> variable:[old value, newvalue]
+		// if variable:[value1, value2] -> variable:[value1, value2, newvalue]
+
 		if !ok {
-			variables[variable] = value // if there is no key, add key&value as string
+			if len(dotKeys) == 0 {
+				variables[variable] = value
+			} else {
+				for i := len(dotKeys[1:]); i > 0; i-- {
+					if i == len(dotKeys[1:]) {
+						variables[variable] = map[string]interface{}{dotKeys[i]: value}
+					} else {
+						variables[variable] = map[string]interface{}{dotKeys[i]: variables[variable]}
+					}
+				}
+			}
 		} else {
 			switch storedValType := storedValue.(type) {
 			case string:
 				variables[variable] = []string{storedValType, value}
 			case []string:
 				variables[variable] = append(storedValType, value)
+			case map[string]interface{}:
+				return nil, fmt.Errorf("(%s) already exists as an exploding param", variable)
```

Comment:
> exploding param

😂 That is awesome but I think "exploding param" may be confusing. What about this?

`was already used as an object path with the dot syntax and cannot be mixed with other formats`

Or something like that. What do you think?

Reply:
That works!